File: language/src/main/java/robok/trindade/terminal/RobokTerminal.java
package com.gam.engine;

import android.view.View;
import android.widget.LinearLayout;
import java.util.List;
import com.gam.engine.R;
import java.util.ArrayList;

public class RobokTerminal {

    private LinearLayout terminal;
    private View bottomSheetView;
    public static List<String> logs;
    
    public static final String WHITE = "#FFFFFF";
    public static final String BLACK = "#000000";
    public static final String ERROR_COLOR = "#FF0000";
    public static final String WARNING_COLOR = "#FFC400";
    public static final String SUCCESS_COLOR = "#198754";
    
    public static final String ERROR_PATTERN = "----------[NUM]. ERROR)";
    public static final String WARNING_PATTERN = "----------[NUM]. WARNING)";
    
	public RobokTerminal(){
		logs = new ArrayList<>();
		
	}
	
    public void addLog(String log) {
        logs.add("\n" + log);
    }
    
    public String getLogs () {
        return logs.toString();
    }
    
    public void addLog(String TAG, String LOG) {
		String tag = TAG.replace("[NUM]","" + (logs.size() + 1));
		
        addLog(tag + " : " + LOG);
    }

    public void addWarningLog(String TAG, String LOG) {
        String tag_temp = ERROR_PATTERN + TAG;
        tag_temp = tag_temp.replace("[NUM]","" + (logs.size() + 1));
		
        addLog(tag_temp + " : " + LOG);
    }

    public void addErrorLog(String TAG, String LOG, int num) {
        String tag_temp = ERROR_PATTERN + TAG;
        tag_temp = tag_temp.replace("[NUM]","" + (logs.size() + 1));
		
        addLog(tag_temp + " : " + LOG);
    }
}

File: app/src/main/java/dev/trindade/robokide/terminal/RobokTerminal.kt
package dev.trindade.robokide.terminal

import android.content.Context
import android.view.LayoutInflater
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatDelegate
import com.google.android.material.bottomsheet.BottomSheetDialog
import dev.trindade.robokide.R

class RobokTerminal(context: Context) : BottomSheetDialog(context) {

    private val terminal: LinearLayout
    private val bottomSheetView = LayoutInflater.from(context).inflate(R.layout.dialog_terminal, null)
    
    init {
        setContentView(bottomSheetView)
        setCancelable(true)
        terminal = bottomSheetView.findViewById(R.id.background_terminal)
    }

    private val isDarkMode: Boolean
        get() = when (AppCompatDelegate.getDefaultNightMode()) {
            AppCompatDelegate.MODE_NIGHT_NO -> false
            AppCompatDelegate.MODE_NIGHT_YES -> true
            else -> false
        }

    companion object {
        const val WHITE = "#FFFFFF"
        const val BLACK = "#000000"
        const val ERROR_COLOR = "#FF0000"
        const val WARNING_COLOR = "#FFC400"
        const val SUCCESS_COLOR = "#198754"
    }
}